parachain/runtime/tests/mock/mod.rs

```diff
@@ -311,139 +361,185 @@ pub fn assert_store_main_chain_header_event(height: u32, hash: H256Le, relayer:
     assert!(events.iter().any(|a| a.event == store_event));
 }
 
-#[allow(dead_code)]
-pub fn generate_transaction_and_mine(
+#[derive(Default, Clone, Debug)]
+pub struct TransactionGenerator {
     address: BtcAddress,
     amount: u128,
     return_data: Option<H256>,
-) -> (H256Le, u32, Vec<u8>, Vec<u8>) {
-    generate_transaction_and_mine_with_script_sig(
-        address,
-        amount,
-        return_data,
-        &[
-            0, 71, 48, 68, 2, 32, 91, 128, 41, 150, 96, 53, 187, 63, 230, 129, 53, 234, 210, 186,
-            21, 187, 98, 38, 255, 112, 30, 27, 228, 29, 132, 140, 155, 62, 123, 216, 232, 168, 2,
-            32, 72, 126, 179, 207, 142, 8, 99, 8, 32, 78, 244, 166, 106, 160, 207, 227, 61, 210,
-            172, 234, 234, 93, 59, 159, 79, 12, 194, 240, 212, 3, 120, 50, 1, 71, 81, 33, 3, 113,
-            209, 131, 177, 9, 29, 242, 229, 15, 217, 247, 165, 78, 111, 80, 79, 50, 200, 117, 80,
-            30, 233, 210, 167, 133, 175, 62, 253, 134, 127, 212, 51, 33, 2, 128, 200, 184, 235,
-            148, 25, 43, 34, 28, 173, 55, 54, 189, 164, 187, 243, 243, 152, 7, 84, 210, 85, 156,
-            238, 77, 97, 188, 240, 162, 197, 105, 62, 82, 174,
-        ],
-    )
+    script: Vec<u8>,
+    confirmations: u32,
+    relayer: [u8; 32],
 }
 
-#[allow(dead_code)]
-pub fn generate_transaction_and_mine_with_script_sig(
-    address: BtcAddress,
-    amount: u128,
-    return_data: Option<H256>,
-    script: &[u8],
-) -> (H256Le, u32, Vec<u8>, Vec<u8>) {
-    let mut height = 1;
-    let confirmations = 6;
-
-    // initialize BTC Relay with one block
-    let init_block = BlockBuilder::new()
-        .with_version(2)
-        .with_coinbase(&address, 50, 3)
-        .with_timestamp(1588813835)
-        .mine(U256::from(2).pow(254.into()))
-        .unwrap();
-
-    let raw_init_block_header =
-        RawBlockHeader::from_bytes(&init_block.header.try_format().unwrap())
-            .expect("could not serialize block header");
+impl TransactionGenerator {
+    pub fn new() -> Self {
+        Self {
+            relayer: ALICE,
+            confirmations: 7,
+            amount: 100,
+            script: vec![
```

Comment:
Not for this PR, but we should definitely have some sort of `Builder` for the scriptSig.